### PR TITLE
Homepage redesign

### PR DESCRIPTION
**Pull request by katjuell**
### What should this PR do?
Resolves [DEVED-136](https://sourcegraph.atlassian.net/browse/DEVED-136) by updating the homepage to organize content around key concepts.

### Why are we making this change?
- Our homepage has displayed all content up until now. This will not scale well as we add more content; it also does not help users navigate different conceptual areas. 
- Our homepage redesign will surface content to users, clustered around concepts that we think are important. It will also be easier to parse visually, since there will be only select elements visible.

### What are the acceptance criteria? 
- Styles and organization look reasonable and make sense.
- ^ On mobile as well.

### How should this PR be tested?
- Check out the branch and test the above criteria.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -1,34 +1,19 @@
...
-import { FunctionComponent } from 'react'
 
-interface Props {
-    posts: MarkdownFileWithUrl[]
-}
-
-export const getStaticProps: GetStaticProps<Props> = async () => {
+export const getStaticProps: GetStaticProps<HomepageTemplateProps> = async () => {
     const posts = await loadAllRecords('posts')
+    const searchPosts = posts.filter(record => record.frontMatter.tags.includes('search')).slice(0,3)
```
Comment (katjuell) [inline pages/index.tsx:8]: Just one way that we can select content for the homepage. Open to feedback!

**Comment by katjuell** [comment]
https://user-images.githubusercontent.com/43454804/131389107-95520116-33a1-45e4-8044-11eeb846a71e.mov

**Comment by katjuell** [comment]
Screen Shot 2021-08-30 at 2 36 41 PM

iphone:
Screen Shot 2021-08-30 at 2 37 07 PM

ipad:
Screen Shot 2021-08-30 at 2 37 20 PM

**Comment by katjuell** [comment]
@ltagliaferri I thought this was a good stopping point to get feedback!

**Comment by ltagliaferri** [comment]
> @ltagliaferri I thought this was a good stopping point to get feedback!

I think this looks really good, I love the animation. I wonder if there is a way we can gesture to there being more content... I know carousels are passé but maybe we can have an arrow or something that links into the tag for `Search` and `Video` respectively?

**Comment by katjuell** [comment]
Yeah, that sounds goo! Would you prefer that to a bottom button
Screen Shot 2021-08-30 at 5 01 28 PM
? Like

**Comment by ltagliaferri** [comment]
Yeah, I think that that sounds good. I guess I would use `tutorials` in there, and not sure how I feel about the "see" language in terms of accessibility, is there another verb we could use that is not as drawn out as "check out" or as formal as "review"?

**Comment by katjuell** [comment]
`Load more tutorials`? Or is that too mechanical? Maybe even `Load more` would be enough

**Comment by ltagliaferri** [comment]
Yeah, let us try that!

File: .eslintrc.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
     "import/no-default-export": "off",
     "react/forbid-dom-props": "warn",
     "@typescript-eslint/require-await": "warn",
-    "react/react-in-jsx-scope": "off"
+    "react/react-in-jsx-scope": "off",
+    "multiline-ternary": "error"
```
Comment (katjuell) [inline .eslintrc.json:10]: Adding this to enhance readability

File: components/Footer/FooterStyles.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ interface Props {
 }
 
 export const StyledFooterWrapper = styled.footer<Props>`
+    margin: 18rem 0 0;
```
Comment (katjuell) [inline components/Footer/FooterStyles.ts:8]: Should we bump this up even more?

**Comment by katjuell** [comment]
Do we want to add another section?

**Comment by ltagliaferri** [comment]
> Do we want to add another section?

I don't think we have enough content right now, but soon!

**Review by ltagliaferri — APPROVED** [review]
This looks great, feel free to add more space to the footer if you would like, I tested locally and can see it both ways

**Pull request merged**